File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static ArrayList<Integer>[] list;
    static boolean[] visited;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int start = Integer.parseInt(st.nextToken());

        visited = new boolean[n+1];
        list = new ArrayList[n+1];
        for(int i=1; i<=n; i++){
            list[i] = new ArrayList<>();
        }

        for(int i=0; i<m; i++){
            st = new StringTokenizer(br.readLine());
            int s = Integer.parseInt(st.nextToken());
            int e = Integer.parseInt(st.nextToken());

            list[s].add(e);
            list[e].add(s);
        }

        for(int i=1; i<=n; i++){
            Collections.sort(list[i]);
        }

        dfs(start);
        System.out.println();
        Arrays.fill(visited, false);
        bfs(start);
    }

    public static void dfs(int start){
        visited[start] = true;
        System.out.print(start + " ");

        for(int x: list[start]){
            if(visited[x]) continue;

            dfs(x);
        }
    }

    public static void bfs(int start){
        Queue<Integer> que = new LinkedList<>();
        que.add(start);
        visited[start] = true;

        while(!que.isEmpty()){
            int s = que.poll();
            System.out.print(s + " ");

            for(int x: list[s]){
                if(!visited[x]) {
                    visited[x] = true;
                    que.add(x);
                }
            }
        }
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const [n, m, v] = input[0].split(' ').map(Number);
let graph = new Array(n+1);

for(let i=0; i <= n; i++){
    graph[i] = [];
}

for(let i=1; i<=m; i++){
    let [from, to] = input[i].split(' ').map(Number);
    graph[from].push(to);
    graph[to].push(from);
}

for(let i=1; i<=n; i++){
    graph[i].sort((a, b) => a- b);
}

let result = [];
let visited = new Array(n+1).fill(false);

function dfs(now){
    result.push(now);

    for(let x of graph[now]){
        if(!visited[x]){
            visited[x] = true;
            dfs(x);
        }
    }
}

function bfs(start){
    let queue = [];
    queue.push(start);

    while(queue.length !== 0){
        let now = queue.shift();
        result.push(now);

        for(let x of graph[now]){
            if(!visited[x]){
                visited[x] = true;
                queue.push(x);
            }
        }
    }
}

visited[v] = true;
dfs(v);
console.log(result.join(' '));

result = [];
visited.fill(false);
visited[v] = true;
bfs(v);
console.log(result.join(' '));